lib/esp32/espnow.toit

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+STATION ::= 0 // Wi-Fi station mode
+SOFTAP  ::= 1 // Not support yet
+
+check_mac_ address/ByteArray:
+  if address.size != 6:
+      throw "ESP-Now MAC address length must be 6 bytes"
+
+check_key_ key/ByteArray:
+  if key.size != 16:
+      throw "ESP-Now key length must be 16 bytes"
+
+check_channel_ channel/int:
+  if not 0 <= channel <= 14:
+      throw "ESP-Now channel range must be 0-14"
+
+class Address:
+  mac/ByteArray
+
+  constructor .mac:
+    check_mac_ mac
+
+  stringify -> string:
+    return "$(%02x mac[0]):$(%02x mac[1]):$(%02x mac[2]):$(%02x mac[3]):$(%02x mac[4]):$(%02x mac[5])"
+
+class Key:
+  data/ByteArray
+
+  constructor .data/ByteArray:
+    check_key_ data
+  
+  constructor.from_string string_data/string:
+    return Key string_data.to_byte_array
+
+class Datagram:
+  address/Address
+  data/ByteArray
+
+  constructor .address .data:
+
+class Service:
+  _group ::= ?
+
+  constructor --mode/int=STATION:
+    _group = espnow_init_ mode #[]
+  
+  constructor.with_key --mode/int=STATION --key/Key:
+    _group = espnow_init_ mode key.data
+
+  send data/ByteArray --address/Address --wait/bool=true -> int:
+    return espnow_send_ address.mac data wait
```

Comment:
Looking through the `send` code, it seems like the function never returns anything else than 0.
-> I would change this to:
```suggestion
  send data/ByteArray --address/Address --wait/bool=true -> none:
    espnow_send_ address.mac data wait
```
If there is an error, it would yield an exception.

Reply:
In the send primitive side, the function `send` may return error code, like `ETIMEDOUT` and `EIO`.

Reply:
The `Primitive::os_error(ETIMEDOUT, process)` is actually nicer than that: it builds an exception with the error message. As such, the primitive will never send anything else than 0.

In fact, it should then change to return the null-object instead, but fundamentally that's not as important.